clients/adapter/src/main/java/org/eclipse/hono/adapter/client/command/InternalCommandConsumer.java

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.adapter.client.command;
+
+import org.eclipse.hono.util.Lifecycle;
+
+/**
+ * A client to receive commands to the internal Command and Control API endpoint provided by protocol adapters.
+ */
+public interface InternalCommandConsumer extends Lifecycle {
```

Comment:
why do we need this in addition to the already existing *org.eclipse.hono.adapter.client.command.CommandConsumer*?

Reply:
The `InternalCommandConsumer` needs to extend `Lifecycle` and it doesn't need the `close(spanContext)` method of the `CommandConsumer` interface.

Reply:
Then why don't we simply use Lifecycle directly? *InternalCommandConsumer* doesn't seem to define any other behavior anyway, does it?

Reply:
It was meant more as a marker interface. But ok, it doesn't provide much benefit here. I've removed it.